### PR TITLE
Simple SMA strategy

### PR DESCRIPTION
**Pull request by metdinov**
Solves #4

### CODE REVIEW

**Comment by metdinov** [comment]
Usage:   
```python
import backtester as bt
from backtester.strategy import SimpleSMA

port = bt.run("data/SPX_2008-2018.csv", strat_class=SimpleSMA, symbol="SPX", period=200)
report = port.create_report()
```

**Comment by metdinov** [comment]
Added logging.   

```shell
DEBUG:root:2009-06-01 00:00:00 - BUY amount: 1060 - SMA: 926.887250 - Spot: 942.870000
DEBUG:root:2009-06-22 00:00:00 - SELL amount: 1060 - SMA: 900.767850 - Spot: 893.040000
DEBUG:root:2009-06-24 00:00:00 - BUY amount: 1051 - SMA: 897.197550 - Spot: 900.940000
DEBUG:root:2009-07-07 00:00:00 - SELL amount: 1051 - SMA: 884.989950 - Spot: 881.030000
DEBUG:root:2009-07-09 00:00:00 - BUY amount: 1049 - SMA: 881.490750 - Spot: 882.680000
DEBUG:root:2009-07-10 00:00:00 - SELL amount: 1049 - SMA: 879.945300 - Spot: 879.130000
```

**Comment by facundoolano** [comment]
That usage (with a little more explanation) could make it to the README

File: backtester/portfolio/portfolio.py [hunk 1]
```diff
@@ -28,11 +28,18 @@ def update_signal(self, signal):
...
 
         (price, direction) = self._get_price(signal)
-        qty = self._get_allocation(signal.strength, price)
+        qty = self._get_allocation(signal, price)
         (current_amount, current_open_price) = self.current_position.get(
             signal.symbol, (0, 0))
-        new_open_price = (current_open_price * current_amount +
-                          direction * price * qty) / (current_amount + qty)
+
+        if qty == 0:
```
Comment (facundoolano) [inline backtester/portfolio/portfolio.py]: this if + elif + else could be its own function

File: backtester/portfolio/simpleportfolio.py [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+        if signal.direction == "SELL":
+            amount, open_price = self.current_position.get(
+                signal.symbol, (0, 0))
+            if amount > 0:
+                logging.debug("%s - SELL amount: %d - SMA: %f - Spot: %f",
+                              self.data_handler.current_date, amount, sma,
+                              price)
+            return amount
+        else:
+            amount = math.floor(self.current_position["Cash"] / price)
```
Comment (facundoolano) [inline backtester/portfolio/simpleportfolio.py]: The uppercase `"Cash"` seems odd

File: backtester/portfolio/simpleportfolio.py [hunk 3]
```diff
@@ -1,13 +1,34 @@
...
 
     def __init__(self, *args):
         super().__init__(*args)
 
-    def _get_allocation(self, strength, price):
+    def _get_allocation(self, signal, price):
         """Allocates all capital to the given signal"""
-        return math.floor(self.current_position["Cash"] / price)
+        win_pct, sma = signal.strength
+        if signal.direction == "SELL":
```
Comment (facundoolano) [inline backtester/portfolio/simpleportfolio.py]: you could handle these as constants: `event.SELL`, `event.BUY`, `event.MARKET_TYPE`, etc.

File: backtester/strategy/simplesma.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        self.events = events
+
+    def generate_signals(self, event):
+        period_bars = self.data_handler.get_latest_bars(
+            self.symbol, N=self.period)
+        if period_bars["ask"].count() < self.period:
+            return
+
+        sma = period_bars["ask"].mean()
+        if period_bars["ask"].iloc[-1] >= sma:
```
Comment (facundoolano) [inline backtester/strategy/simplesma.py]: Out of curiosity, do you always have to either buy or sell? you never hold?

**Comment by facundoolano** [comment]
Also you could add a unit test for the new strategy

**Pull request closed without merging**